### PR TITLE
ncm-gpfs: reinstall gpfs update packages after base

### PR DESCRIPTION
**Pull request by kwaegema**
the gpfs component removes the update packages and installs the base packages, but didn't reinstall the update packages, therefore, this is not installed correctly in the ks-post-reboot phase

### CODE REVIEW

File: ncm-gpfs/src/main/perl/gpfs.pm [hunk 1]
```diff
@@ -265,6 +263,26 @@ sub runcurl {
...
+sub reinstall_update_rpms {
+    my ($self, $config, $rpms) = @_;
+    my $ret = 1;
+    my $useyum =  $config->getValue("$mypath/base/useyum");
+    my @reinrpms = @$rpms;
+
+    if (scalar @reinrpms) {
+        if ($useyum) {
+            # for dependencies
+            $self->runyum($config, "install", @reinrpms) || return;
```
Comment (stdweird) [inline ncm-gpfs/src/main/perl/gpfs.pm]: don't return here. for both `runyum` and `runcmd`, do a `$ret = $self->run...`; and end this block with
```perl
if($ret) {
    $self->info("Rpms reinstalled");
} else {
   $self->error("Reinstalling rpms failed");
}
```

File: ncm-gpfs/src/main/perl/gpfs.pm [hunk 2]
```diff
@@ -265,6 +263,26 @@ sub runcurl {
     return $output || 1;
 };
 
+sub reinstall_update_rpms {
+    my ($self, $config, $rpms) = @_;
+    my $ret = 1;
+    my $useyum =  $config->getValue("$mypath/base/useyum");
+    my @reinrpms = @$rpms;
+
+    if (scalar @reinrpms) {
```
Comment (stdweird) [inline ncm-gpfs/src/main/perl/gpfs.pm]: `scalar` is not needed

File: ncm-gpfs/src/main/perl/gpfs.pm (hunk 1 above)
Comment (stdweird) [inline ncm-gpfs/src/main/perl/gpfs.pm]: maybe also fix this in `remove_existing_rpms`

**Comment by stdweird** [comment]
@kwaegema some unittests would also be nice. i'm tagging this 16.10 due to feature freeze (and we're probably the only ones using this anyway)

**Comment by stdweird** [comment]
@kwaegema squash the commit, also add unittests

**Comment by jrha** [comment]
@kwaegema any response to @piojo's comments?

**Comment by stdweird** [comment]
@kwaegema can you cleanup the commit messages? (at least prefix `ncm-gpfs`, and try to squash the remarks one if possible)

**Comment by kwaegema** [comment]
This is now rebased on #911 to use same test resources. Please review/merge #911 first before this one

**Comment by jrha** [comment]
@piojo any chance you can review this again now that it has been updated?

**Comment by stdweird** [comment]
@kwaegema can you rebase on master? i'll review again after that

**Comment by kwaegema** [comment]
@stdweird rebased

**Pull request merged**